### ISSUE TITLE
Optimize rendering of cbx files

### ISSUE THREAD

**Issue by kjk** [sumatrapdfreader/sumatrapdf#3411]
Investigation triggered by https://github.com/sumatrapdfreader/sumatrapdf/discussions/3365#discussioncomment-5479431

 We create 2 bitmaps, which doubles rendering time and memory usage.

In `EngineImages` we create a first bitmap by decoding an image into `gdiplus::Bitmap`, store it in `ImagePage`.

Then in `EngineImages::RenderPage` we create `HBITMAP` and return as `RenderedBitmap`.

`RenderedBitmap` is then kept in cache as well as every bitmap in `ImagePage`.

We need 2 changes.

First, `RenderedBitmap` is a bad abstraction as it bakes the physical format of the bitmap as `HBITMAP` which requires us to convert from other formats (mupdf's pixmap, Gdiplus:Bitmap).

What we really need is an interface that can blit to HDC.

We also need `Drop` function for the cache to release the bitmap, probably by refcounting.

Plan of actions:
* [x] define a new abstract base class e.g. `BlittableBitmap` with `Drop()` and `Blit(HDC, x, y, dx, dy)`
* [x] change `RenderedBitmap` to inherit from that and change how it's rendered by `Blit`. Possibly rename to `BlittableHBITMAP`
* [x] optimize `EngineImages:RenderPage` by returning `BittableGdiplusBitmap` that wraps `Gdiplus::Bitmap`
* [x] don't cache `Gdiplus::Bitmap` in `ImagePage::bmp` as it's redundant with render cache
* [x] look into optimizing mupdf usage by using `BlittablePixmap` that wraps mupdf `pixmap`

**Comment by kjk** [comment]
Make `BlittableImage` ref-counted so that can participate in caching.

Remove `ImagePage` and `EngineImages::pageCache`. add `BlittableImage` to `ImagePageInfo` so that we can ref count it.

**Comment by GitHubRulesOK** [comment]
HMM there may be clues from another's perspective https://github.com/sumatrapdfreader/sumatrapdf/compare/master...ftk:sumatrapdf:master

**Comment by kjk** [comment]
Thanks for the pointer. Will study to see what we can lift from there.

Funnily enough, the biggest memory saving change they did is the same thing I did in my change.

**Comment by kjk** [comment]
## Next step (agent instructions for future work)

Status: the expensive part of this issue is already done. `ImagePage` now holds at most one decoded form (`img` = lazy compressed mupdf `fz_image`, or `bmp` = `Gdiplus::Bitmap`), and `EngineImages::RenderPage` has a mupdf fast path that decodes JPEGs at near-target DCT scale (1, 1/2, 1/4, 1/8). The unconditional "two full-size bitmaps cached" is gone for the common path.

What remains: the last unchecked plan item — a `BlittablePixmap` that wraps `fz_pixmap` and blits directly — and the GDI+ fallback double.

### Concrete measurement (so future work is gated on proof)
- Do NOT use working set (peak WS gave 449 vs 447 vs 314 MB for near-identical runs — pure noise).
- Use **private committed bytes**, ~4 s settle, baseline-subtracted. Launch `SumatraPDF-dll.exe -for-testing -zoom <z> <file>`, sample `PrivateMemorySize64`, subtract the `-for-testing` (no file) baseline.
- Baseline observed: 32 MB. For a 5000x4000 image (~80 MB decoded BGRA):
  - JPEG/PNG fit-page: ~+76 MB
  - JPEG 100%: ~+132 MB (~1.7x), PNG 100%: ~+151 MB (~1.9x)
- => at full-res a rendered page retains ~1.7-1.9x one bitmap. Likely the mupdf decoded pixmap (kept in `fz_store`) + our DIB copy. That's the ~30-45%/page opportunity.

### The task
1. First, instrument to confirm the second copy is OURS to remove (not just `fz_store`): log bytes allocated by the DIB in `FzPixmapToRenderedBitmap` (EngineImages.cpp ~L300) vs the `fz_store` size after a render. If the duplicate is the store, this change won't help — stop.
2. If confirmed: add `BlittablePixmap : BlittableBitmap` (wrap an `fz_pixmap`, `Blit()` via `StretchDIBits`/`SetDIBitsToDevice` with a top-down `BITMAPINFO`, `Drop()` via `fz_drop_pixmap` on the engine's per-thread `fz_context`). Have the `RenderPage` mupdf fast path return it instead of converting pixmap -> DIB. Keep the BGR conversion (GDI needs BGRA byte order) but skip the `CreateDIBSection` + memcpy.
3. Acceptance gate: before/after private-bytes on a real `.cbz` must show a clear drop (turn ~1.7x into ~1x). No measurable drop => don't merge.

### Skip
- `BlittableGdiplusBitmap` for the GDI+ fallback path (rotation/tiles/multi-frame TIFF/EXIF) — uncommon; not worth the refcounting complexity.
- Mind the existing concurrency model (per-thread cloned `fz_context`, `drawLock`, `ImagePage::refs`): a `BlittablePixmap` shared between the render cache and a worker must keep the `fz_pixmap` alive via mupdf refcount, and only touch mupdf through a context owned by the calling thread.

**Comment by kjk** [comment]
this is done
